src/debug/ee/controller.cpp

```diff
@@ -7938,15 +7894,24 @@ void DebuggerJMCStepper::TriggerMethodEnter(Thread * thread,
     if (!fIsUserCode)
         return;
 
+    if ((m_eMode != cStepIn) && (m_fp != LEAF_MOST_FRAME))
+    {
+        // We can run a stack trace in managed code (eg, not a stub)
+        ControllerStackInfo info;
+        StackTraceTicket ticket(ip);
+        info.GetStackInfo(ticket, thread, LEAF_MOST_FRAME, NULL);
+
+        if (IsCloserToLeaf(info.m_activeFrame.fp, m_fp))
```

Comment:
This works for `cStepOver`.  I suspect we may need to handle the `cStepOut` case separately.